Remotion/ObjectBinding/Web/UI/Controls/Validation/BocListValidationResultDispatchingValidator.cs

```diff
@@ -200,5 +194,34 @@ private BocList GetControlToValidate ()
 
       return bocListControl;
     }
+
+    void IValidatorWithDynamicErrorMessage.RefreshErrorMessage ()
+    {
+      var bocList = GetControlToValidate();
+      var validationRepository = bocList.ValidationFailureRepository;
+
+      var sb = new StringBuilder();
+
+      var validationFailuresForBusinessObjectPropertyOfBocList = validationRepository.GetUnhandledValidationFailuresForBocList(true);
+      foreach (var validationFailure in validationFailuresForBusinessObjectPropertyOfBocList)
+      {
+        sb.AppendLine(validationFailure.Failure.ErrorMessage);
+      }
+
+      var hasRowsWithUnhandledValidationFailures = validationRepository.GetUnhandledValidationFailuresForBocListAndContainingDataRowsAndDataCells(true).Any();
+      if (hasRowsWithUnhandledValidationFailures)
+      {
+        var remainingValidationFailuresText =bocList
+            .GetResourceManager()
+            .GetString(BocList.ResourceIdentifier.RemainingValidationFailureText);
+
+        sb.AppendLine(remainingValidationFailuresText);
+      }
+
+      if (validationFailuresForBusinessObjectPropertyOfBocList.Any() || hasRowsWithUnhandledValidationFailures)
```

Comment:
If validationRepository.GetUnhandledValidationFailuresForBocList(true); returns an enumerable, we would trigger a second enumeration here. given the enture setup, the following might be best (and more expressive):
```suggestion
      if (errorMessage.Length > 0)
```